src/helper/account_checks.rs

```diff
@@ -0,0 +1,41 @@
+use pinocchio::{
+    account_info::AccountInfo,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+use crate::helper::utils::DataLen;
+
+#[inline(always)]
+pub fn check_signer(account: &AccountInfo) -> Result<(), ProgramError> {
+    if !account.is_signer() {
+        return Err(ProgramError::MissingRequiredSignature);
+    }
+    Ok(())
+}
+
+#[inline(always)]
+pub fn check_ix_payer_valid(account: &AccountInfo, ix_payer: &Pubkey) -> Result<(), ProgramError> {
```

Comment:
i dont think we need this fn. it just takes an arbitary acc and compares pubkey arg.